src/DotVVM.Framework/Resources/Scripts/global-declarations.ts

```diff
@@ -8,85 +8,121 @@ type DotvvmPostbackHandler = {
     after?: Array<string | DotvvmPostbackHandler>
     before?: Array<string | DotvvmPostbackHandler>
 }
-type PostbackRejectionReason =
-    | { type: "handler", handlerName: string, message?: string }
+
+type DotvvmPostbackErrorLike = {
+    readonly reason: DotvvmPostbackErrorReason
+}
+
+type DotvvmPostbackErrorReason =
+    | { type: 'handler', handlerName: string, message?: string }
     | { type: 'network', err?: any }
     | { type: 'gate' }
     | { type: 'commit', args?: DotvvmErrorEventArgs }
     | { type: 'csrfToken' }
     | { type: 'serverError', status?: number, responseObject: any, response?: Response }
     | { type: 'event' }
+    | { type: 'validation', responseObject: any, response?: Response }
     & { options?: PostbackOptions }
 
-interface AdditionalPostbackData {
-    [key: string]: any
-    validationTargetPath?: string
-}
+type PostbackCommandType = "postback" | "staticCommand" | "spaNavigation"
 
 type PostbackOptions = {
-    readonly additionalPostbackData: AdditionalPostbackData
     readonly postbackId: number
-    readonly sender?: HTMLElement
+    readonly commandType: PostbackCommandType
     readonly args: any[]
+    readonly sender?: HTMLElement
     readonly viewModel?: any
+    serverResponseObject?: any
+    validationTargetPath?: string
 }
 
-type PostbackEventArgs = DotvvmEventArgs & {
-    readonly postbackClientId: number
-    readonly sender?: Element
-    readonly xhr?: XMLHttpRequest
-    readonly serverResponseObject?: any
-    readonly postbackOptions: PostbackOptions
-}
-
-type DotvvmEventArgs = {
-    /** The global view model */
-    readonly viewModel?: any
-}
-
-type DotvvmErrorEventArgs = {
-    readonly sender?: Element
-    readonly serverResponseObject?: any
-    readonly viewModel?: any
-    readonly isSpaNavigationError?: true
+type DotvvmErrorEventArgs = PostbackOptions & {
+    readonly response?: Response
+    readonly error: DotvvmPostbackErrorLike
     handled: boolean
 }
 
-type DotvvmBeforePostBackEventArgs = PostbackEventArgs & {
+type DotvvmBeforePostBackEventArgs = PostbackOptions & {
     cancel: boolean
 }
-type DotvvmAfterPostBackEventArgs = PostbackEventArgs & {
-    handled: boolean
+
+type DotvvmAfterPostBackEventArgs = PostbackOptions & {
     /** Set to true in case the postback did not finish and it was cancelled by an event or a postback handler */
     readonly wasInterrupted: boolean;
```

Comment:
I'd chage it to `wasInterrupted?: true`, so we don't have to include it on all the places where it's false.